pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -208,11 +213,16 @@ func (s *fileSSTSink) write(ctx context.Context, resp exportedSpan) error {
 
 	span := resp.metadata.Span
 
+	spanPrefix, err := elidedPrefix(span.Key, s.elideMode)
+	if err != nil {
+		return err
+	}
+
 	// If this span starts before the last buffered span ended, we need to flush
 	// since it overlaps but SSTWriter demands writes in-order.
 	if len(s.flushedFiles) > 0 {
 		last := s.flushedFiles[len(s.flushedFiles)-1].Span.EndKey
-		if span.Key.Compare(last) < 0 {
+		if span.Key.Compare(last) < 0 || !bytes.Equal(spanPrefix, s.elidePrefix) {
```

Comment:
could you add a test case to file_sink_test.go that validates that two's sst's with different prefixes get flushed to different ssts?

Reply:
it may also be worth adding an outer loop to that that so that all test cases deal with/without stripping prefixes.

Reply:
Done